graylog2-web-interface/src/components/maps/configurations/GeoIpResolverConfig.tsx

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useEffect, useState } from 'react';
+import { Field, Form, Formik } from 'formik';
+
+import { IfPermitted, Select } from 'components/common';
+import { Button, Col, Input, Modal, Row } from 'components/bootstrap';
+import FormikInput from 'components/common/FormikInput';
+import { DocumentationLink } from 'components/support';
+
+export type GeoDatabaseType = 'MAXMIND' | 'IPINFO'
+
+export type GeoIpConfigType = {
+  enabled: boolean;
+  enforce_graylog_schema: boolean;
+  db_vendor_type: GeoDatabaseType;
+  city_db_path: string;
+  asn_db_path: string;
+  run_before_extractors: boolean;
+}
+
+export type OptionType = {
+  value: string;
+  label: string;
+}
+
+type Props = {
+  config: GeoIpConfigType,
+  updateConfig: (config: GeoIpConfigType) => Promise<GeoIpConfigType>,
+};
+
+const defaultConfig: GeoIpConfigType = {
+  enabled: false,
+  enforce_graylog_schema: true,
+  db_vendor_type: 'MAXMIND',
+  city_db_path: '/etc/graylog/server/GeoLite2-City.mmdb',
+  asn_db_path: '/etc/graylog/server/GeoLite2-ASN.mmdb',
+  run_before_extractors: false,
+};
+
+const GeoIpResolverConfig = ({ config = defaultConfig, updateConfig }: Props) => {
+  const [showModal, setShowModal] = useState(false);
+  const [curConfig, setCurConfig] = useState(() => {
+    return { ...defaultConfig };
+  });
+
+  useEffect(() => {
+    setCurConfig({ ...config });
+  }, [config]);
+
+  const resetConfig = () => {
+    setCurConfig(config);
+    setShowModal(false);
+  };
+
+  const handleSubmit = (values) => {
+    updateConfig(values)
```

Comment:
I have only one optional improvement in mind. When we return the promise here, we can use Formiks loading state, to disable the save button while the request is not finished.

Reply:
There is no need to call `setSubmitting` manually, you can also just return the promise (`updateConfig `).